src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1439,11 +1737,18 @@ private void VisitObjectElementInitializer(int containingSlot, BoundAssignmentOp
                         {
                             VisitArguments(objectInitializer, objectInitializer.Arguments, objectInitializer.ArgumentRefKindsOpt, (PropertySymbol)symbol, objectInitializer.ArgsToParamsOpt, objectInitializer.Expanded);
                         }
+
                         if ((object)symbol != null)
                         {
                             int slot = (containingSlot < 0) ? -1 : GetOrCreateSlot(symbol, containingSlot);
                             VisitObjectCreationInitializer(symbol, slot, node.Right);
+                            // PROTOTYPE(nullable-api): Should likely be setting _resultType in VisitObjectCreationInitializer
+                            // and using that value instead of reconstructing here
                         }
+
+                        var result = new VisitResult(objectInitializer.Type, NullableAnnotation.NotAnnotated, NullableFlowState.NotNull);
+                        SetAnalyzedNullability(objectInitializer, result);
+                        SetAnalyzedNullability(node, result);
```

Comment:
Are these necessary?

Reply:
Yes. The entire handling of object initializers does not go through `VisitExpressionWithoutStackGuard`, so we need to set them here.

---
In reply to: [265702557](https://github.com/dotnet/roslyn/pull/34017#discussion_r265702557) [](ancestors = 265702557)

Reply:
Are we setting values in the map other than `node.Type` though? In short, could the consumer of the map simply treat all missing values as the default rather than requiring all expressions to be visited in `NullableWalker` and added to the map?

---
In reply to: [265704027](https://github.com/dotnet/roslyn/pull/34017#discussion_r265704027) [](ancestors = 265704027,265702557)

Reply:
Yes. This node is a `BoundAssignmentOperator`: if it's a nullable lvalue that's set a non-null rvalue that should be reflected in the public API.

---
In reply to: [265707252](https://github.com/dotnet/roslyn/pull/34017#discussion_r265707252) [](ancestors = 265707252,265704027,265702557)

Reply:
To be clear: currently, it's just setting `NotAnnotated`/`NotNull`. In reality, we need to adjust the implementation of the `VisitObjectInitializer` function to do the right thing, and then set the assignment to have the same result type.

---
In reply to: [265708722](https://github.com/dotnet/roslyn/pull/34017#discussion_r265708722) [](ancestors = 265708722,265707252,265704027,265702557)